### PR TITLE
Propagate configuration updates automatically to clients

### PR DESCRIPTION
**Pull request by cheng-chang**
This PR merges the feature branch into master.

### CODE REVIEW

**Comment by cheng-chang** [comment]
@madanadit @ggezer PTAL

**Comment by madanadit** [comment]
@cheng-chang is there anything in particular you need a review on? all commits to the feature branch have been reviewed before?

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -2678,6 +2696,7 @@ public String toString() {
           .setDescription("The maximum number of workers to retry before the client gives up on "
               + " reading a block")
           .build();
+  // TODO(cc): remove this since configuration propagation is always enabled?
```
Comment (madanadit) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java]: did we decide to use a single property for refresh mode: never, once or using heartbeat?

Comment (cheng-chang) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: for path conf, if the mode is never or once, the path conf feature is of no use to clients?

Comment (madanadit) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: makes sense

**Comment by madanadit** [comment]
alluxio-bot, feature-merge this please

**Pull request merged**